packages/web-components/src/search-widget/client/components/pagination.svelte

```diff
@@ -0,0 +1,92 @@
+<script>
+  import { onDestroy } from 'svelte'
+  import { getProperties } from '../api/properties'
+  import { getPropertyImages } from '../api/property-images'
+  import searchWidgetStore from '../core/store'
+
+  let totalPage = 1
+  let pageNumber = 1
+
+  const unsubscribeSearchWidgetStore = searchWidgetStore.subscribe(store => {
+    totalPage = store.totalPage
+    pageNumber = store.pageNumber
+  })
+
+  onDestroy(() => {
+    unsubscribeSearchWidgetStore()
+  })
+
+  const handlePageChangeClick = (pageNumber) => async() => {
```

Comment:
Why not set the current page on the store, then call fetchProperties in the onUpdate hook in the search form - would mean we don't have to copy paste the fetching of properties and images in here and we could re-use the paginator elsewhere

Reply:
@willmcvay so do you mean https://svelte.dev/docs#afterUpdate and https://svelte.dev/docs#beforeUpdate? So I think cannot use store.update in these hooks, because it'll cause infinity re-render for the component. So do you have another idea?

Reply:
@willmcvay fixed as your comment